xbmc/guilib/test/TestGUIControlFactory.cpp

```diff
@@ -0,0 +1,665 @@
+                                            value,
+                                            min);
+  }
+
+  static bool GetDimensions(const TiXmlNode* node,
+                            const char* leftTag,
+                            const char* rightTag,
+                            const char* centerLeftTag,
+                            const char* centerRightTag,
+                            const char* widthTag,
+                            const float parentSize,
+                            float& left,
+                            float& width,
+                            float& min_width)
+  {
+    return CGUIControlFactory::GetDimensions(node,
+                                             leftTag,
+                                             rightTag,
+                                             centerLeftTag,
+                                             centerRightTag,
+                                             widthTag,
+                                             parentSize,
+                                             left,
+                                             width,
+                                             min_width);
+  }
+
+  static bool GetMovingSpeedConfig(const TiXmlNode* pRootNode,
+                                   const char* strTag,
+                                   UTILS::MOVING_SPEED::MapEventConfig& movingSpeedCfg)
+  {
+    return CGUIControlFactory::GetMovingSpeedConfig(pRootNode,
+                                                    strTag,
+                                                    movingSpeedCfg);
+  }
+
+  static bool GetConditionalVisibility(const TiXmlNode* control,
+                                       std::string& condition,
+                                       std::string& allowHiddenFocus)
+  {
+    return CGUIControlFactory::GetConditionalVisibility(control,
+                                                        condition,
+                                                        allowHiddenFocus);
+  }
+
+  static bool GetString(const TiXmlNode* pRootNode,
+                        const char* strTag,
+                        std::string& strString)
+  {
+    return CGUIControlFactory::GetString(pRootNode,
+                                         strTag,
+                                         strString);
+  }
+};
+
+namespace {
+
+using namespace std::string_literals;
+
+struct AlignmentTest
+{
+  std::string def;
+  uint32_t align;
+  bool result = true;
+};
+
+const auto AlignmentTests = std::array{
+    AlignmentTest{R"(<root><test>right</test</root>)"s, XBFONT_RIGHT},
+    AlignmentTest{R"(<root><test>bottom</test</root>)"s, XBFONT_RIGHT},
+    AlignmentTest{R"(<root><test>center</test</root>)"s, XBFONT_CENTER_X},
+    AlignmentTest{R"(<root><test>justify</test</root>)"s, XBFONT_JUSTIFIED},
+    AlignmentTest{R"(<root><test>left</test</root>)"s, XBFONT_LEFT},
+    AlignmentTest{R"(<root><test>foo</test</root>)"s, XBFONT_LEFT},
```

Comment:
each of these is missing the > for the closing test element

```
    AlignmentTest{R"(<root><test>right</test></root>)"s, XBFONT_RIGHT},
    AlignmentTest{R"(<root><test>bottom</test></root>)"s, XBFONT_RIGHT},
    AlignmentTest{R"(<root><test>center</test></root>)"s, XBFONT_CENTER_X},
    AlignmentTest{R"(<root><test>justify</test></root>)"s, XBFONT_JUSTIFIED},
    AlignmentTest{R"(<root><test>left</test></root>)"s, XBFONT_LEFT},
    AlignmentTest{R"(<root><test>foo</test></root>)"s, XBFONT_LEFT},
```

Reply:
fixed.